RemoteSpiNNaker/RemoteSpiNNakerWeb/src/main/java/uk/ac/manchester/cs/spinnaker/jobmanager/JobManager.java

```diff
@@ -202,6 +202,35 @@
      */
     private final Map<Integer, ObjectNode> jobProvenance = new HashMap<>();
 
+    /**
+     * Job ID -> Log messages to be updated.
+     *
+     * Note this will be copied and replaced during execution.
+     */
+    private Map<Integer, String> logsToUpdate = new HashMap<>();
+
+    /**
+     * The action to take when the log update happens.
+     *
+     * Use this to synchronise on log updates actually happening.
+     */
+    private final ActionListener logTimerAction = new ActionListener() {
```

Comment:
This could be a lambda as you're using the Java 8 language profile:
```java
private final ActionListener logTimerAction = (event -> sendLogs());
```
(Outer parentheses may be optional. The style will know.)